jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudwatch/logs/OpenLogStreamInEditorTest.kt

```diff
@@ -0,0 +1,110 @@
+    @JvmField
+    @Rule
+    val projectRule = ProjectRule()
+
+    @JvmField
+    @Rule
+    val mockClientManagerRule = MockClientManagerRule(projectRule)
+
+    @After
+    fun after() {
+        val editorManager = FileEditorManager.getInstance(projectRule.project)
+        runInEdt {
+            editorManager.openFiles.forEach { editorManager.closeFile(it) }
+        }
+    }
+
+    @Test
+    fun testOpeningFileFromGroup() {
+        val cloudWatchMock = delegateMock<CloudWatchLogsClient>()
+        mockClientManagerRule.manager().register(CloudWatchLogsClient::class, cloudWatchMock)
+        whenever(cloudWatchMock.getLogEventsPaginator(Mockito.any<GetLogEventsRequest>()))
+            .thenReturn(object : GetLogEventsIterable(cloudWatchMock, null) {
+                override fun iterator() = mutableListOf(
+                    GetLogEventsResponse.builder().events(
+                        OutputLogEvent.builder().message("abc").build(),
+                        OutputLogEvent.builder().message("def").build()
+                    ).build()
+                ).iterator()
+            })
+
+        val tableModel = ListTableModel<LogStream>(LogStreamsStreamColumn())
+        val table = JBTable(tableModel)
+        tableModel.addRow(LogStream.builder().logStreamName("54321").build())
+        // select the first row
+        table.setRowSelectionInterval(0, 0)
+        val action = OpenLogStreamInEditor(projectRule.project, "12345", table)
+        action.actionPerformed(TestActionEvent())
+        runBlocking {
+            blockUntilFileOpen()
+        }
+        val openFile = FileEditorManager.getInstance(projectRule.project).openFiles.first()
+        assertThat(openFile).isInstanceOf(LightVirtualFile::class.java)
+        assertThat(openFile.isWritable).isFalse()
+        assertThat((openFile as LightVirtualFile).content).isEqualTo("abc\ndef\n")
+    }
+
+    @Test
+    fun testOpeningFileFromStream() {
+        val tableModel = ListTableModel<OutputLogEvent>()
+        tableModel.addRows(
+            mutableListOf(
+                OutputLogEvent.builder().message("abc\n").build(),
+                OutputLogEvent.builder().message("def\n").build(),
+                OutputLogEvent.builder().message("ghi\n").build()
+            )
+        )
+        val action = OpenCurrentInEditor(projectRule.project, "12345", tableModel)
+        action.actionPerformed(TestActionEvent())
+        runBlocking {
+            blockUntilFileOpen()
+        }
+        val openFile = FileEditorManager.getInstance(projectRule.project).openFiles.first()
+        assertThat(openFile).isInstanceOf(LightVirtualFile::class.java)
+        assertThat(openFile.isWritable).isFalse()
+        assertThat((openFile as LightVirtualFile).content).isEqualTo("abc\ndef\nghi\n")
+    }
+
+    private suspend fun blockUntilFileOpen() {
+        val editorManager = FileEditorManager.getInstance(projectRule.project)
+        withTimeout(4000) {
```

Comment:
There is a junit4 rule for coroutines timeouts

https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-debug/src/junit4/CoroutinesTimeout.kt

Reply:
Cool, switched to using that